Source/Core/Core/Core.cpp

```diff
@@ -772,10 +779,16 @@ void SaveScreenShot(std::string_view name)
 
 static bool PauseAndLock(Core::System& system, bool do_lock, bool unpause_on_unlock)
 {
-  // WARNING: PauseAndLock is not fully threadsafe so is only valid on the Host Thread
+  if (do_lock)
+    s_core_mutex.lock();
 
   if (!IsRunning(system))
+  {
+    if (do_lock)
+      s_core_mutex.unlock();
```

Comment:
I think the mutex usage would be cleaner if you used a `std::unique_lock` to eliminate the `unlock` calls.

Reply:
I can't use `std::unique_lock`. If `do_lock` is true and the core is running, then I want to return from this function *with the lock still held*. It will be unlocked later by this function being called with `do_lock` set to false.

Reply:
Ah, yeah. It's not so fun to follow (existing problem). Maybe the function should accept and lock/unlock an existing `unique_lock` then? But maybe that's something for a another PR.

Reply:
Actually, now that I think about it, we shouldn't unlock here if the core isn't running. It should stay locked so that nobody comes and calls Init under our feet.

Reply:
But yeah, if we want to return a `unique_lock` or something along those lines, I would like to leave that for a separate PR.